local/compose/build.go

```diff
@@ -127,57 +157,54 @@ func (s *composeService) ensureImagesExists(ctx context.Context, project *types.
 		}
 
 	}
+	return opts, imagesToBuild, nil
 
-	mode := progress.PrinterModeAuto
-	if quietPull {
-		mode = progress.PrinterModeQuiet
-	}
+}
 
-	err := s.build(ctx, project, opts, observedState, mode)
-	if err == nil {
-		if len(imagesToBuild) > 0 {
-			utils.DisplayScanSuggestMsg()
+func (s *composeService) getImageDigests(ctx context.Context, project *types.Project) (map[string]string, error) {
+	imageNames := []string{}
+	for _, s := range project.Services {
+		imgName := getImageName(s, project.Name)
+		if !utils.StringContains(imageNames, imgName) {
```

Comment:
could use the `map[string]` you just create few lines later to check already resolved image